src/Sentry.AspNetCore/SentryTracingMiddleware.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Sentry.AspNetCore.Extensions;
+using Sentry.Protocol;
+
+namespace Sentry.AspNetCore
+{
+    /// <summary>
+    /// Sentry tracing middleware for ASP.NET Core
+    /// </summary>
+    internal class SentryTracingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly Func<IHub> _hubAccessor;
+
+        public SentryTracingMiddleware(RequestDelegate next, Func<IHub> hubAccessor)
+        {
+            _next = next;
+            _hubAccessor = hubAccessor;
+        }
+
+        /// <summary>
+        /// Handles the <see cref="HttpContext"/>.
+        /// </summary>
+        public async Task InvokeAsync(HttpContext context)
+        {
+            var hub = _hubAccessor();
+            if (!hub.IsEnabled)
+            {
+                await _next(context).ConfigureAwait(false);
+                return;
+            }
+
+            var transaction = hub.CreateTransaction(
+                context.GetTransactionName(),
+                "http.server"
+            );
+
+            try
+            {
+                await _next(context).ConfigureAwait(false);
+            }
+            finally
```

Comment:
If an exception bubbles out of there, is the error in Sentry connected to this transaction?
Sounds like we're missing this connection (the error event should get the trace context data).

It was done in Java recently: https://github.com/getsentry/sentry-java/pull/1156

Raised an issue so we can track this on a follow up PR: https://github.com/getsentry/sentry-dotnet/issues/738

Reply:
No, we don't support this currently. I also mentioned it here: https://github.com/getsentry/sentry-dotnet/issues/663#issuecomment-760227782